### PR TITLE
[docs] removing docs referring ray client.

### LINKED ISSUE #31785: [Core] ray client docs clean up
### What happened + What you expected to happen

remove all the ray_client reference in Ray docs.

### Versions / Dependencies

n/a

### Reproduction script

n/a

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by scv119**
## Why are these changes needed?

Deprecating ray client related docs.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/key-concepts.rst [hunk 1]
```diff
@@ -74,7 +74,6 @@ There are three ways to run a Ray job on a Ray cluster:
 
 1. (Recommended) Submit the job using the :ref:`Ray Jobs API <jobs-overview>`.
 2. Run the driver script directly on any node of the Ray cluster, for interactive development.
-3. Use :ref:`Ray Client <ray-client-ref>` to connect remotely to the cluster within a driver script.
```
Comment (richardliaw) [inline doc/source/cluster/key-concepts.rst:77]: I think you can keep this but say "For Experts only"

File: doc/source/cluster/running-applications/job-submission/index.md [hunk 2]
```diff
@@ -29,10 +29,7 @@ If needed, the Ray Jobs API also provides APIs for [programmatic job submission]
 
 ## Running Jobs Interactively
 
-If you would like to run an application *interactively* and see the output in real time (for example, during development or debugging), you can:
-
-- (Recommended) Run your script directly on a cluster node (e.g. after SSHing into the node using [`ray attach`](ray-attach-doc)), or
-- use [Ray Client](ray-client-ref) to run a script from your local machine while maintaining a connection to the cluster.
```
Comment (richardliaw) [inline doc/source/cluster/running-applications/job-submission/index.md:35]: I think you can keep this but say "For Experts only"

**Review by richardliaw — CHANGES_REQUESTED** [review]
Let's keep the references I commented on, and the rest we should remove.

**Comment by richardliaw** [comment]
BTW, we also need to make sure any explicit instructions/warnings/logs/references to "ray://localhost:10001" are cleaned up in the ray codebase as well; we don't want to give any instructions for users to use ray client on accident.

File: doc/source/cluster/running-applications/job-submission/index.md (hunk 2 above)
Comment (scv119) [reply doc/source/cluster/running-applications/job-submission/index.md:35]: should we keep the ray-client page then?

**Comment by richardliaw** [comment]
yeah just keep the page, right (but reduce general count of references, as
you have done for the rest of the pr)

File: doc/source/ray-observability/ray-logging.rst [hunk 3]
```diff
@@ -170,7 +170,6 @@ Here's a Ray log directory structure. Note that ``.out`` is logs from stdout/std
   This is the log file of the agent containing logs of create/delete requests and cache hits and misses.
   For the logs of the actual installations (including e.g. ``pip install`` logs), see the ``runtime_env_setup-[job_id].log`` file (see below).
 - ``runtime_env_setup-[job_id].log``: Logs from installing :ref:`runtime environments <runtime-environments>` for a task, actor or job.  This file will only be present if a runtime environment is installed.
-- ``runtime_env_setup-ray_client_server_[port].log``: Logs from installing :ref:`runtime environments <runtime-environments>` for a job when connecting via :ref:`Ray Client <ray-client-ref>`.
```
Comment (architkulkarni) [inline doc/source/ray-observability/ray-logging.rst:173]: If we end up leaving Ray Client in as "for experts only", we should maybe also leave this line in to explain what these files are.

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
tests failing

**Comment by zhe-thoughts** [comment]
@cadedaniel @scv119 let's CP it?

**Pull request merged**